clients/include/testing_iamax_iamin_strided_batched.hpp

```diff
@@ -0,0 +1,241 @@
+
+    EXPECT_ROCBLAS_STATUS(func(nullptr, N, dx, incx, incx * N, batch_count, &h_rocblas_result),
+                          rocblas_status_invalid_handle);
+}
+
+template <typename T, void CBLAS_FUNC(rocblas_int, const T*, rocblas_int, rocblas_int*)>
+void template_testing_iamax_iamin_strided_batched(const Arguments&                         arg,
+                                                  rocblas_iamax_iamin_strided_batched_t<T> func)
+{
+
+    rocblas_int N = arg.N, incx = arg.incx, batch_count = arg.batch_count;
+
+    rocblas_stride stridex = arg.stride_x;
+
+    double rocblas_error_1, rocblas_error_2;
+
+    rocblas_local_handle handle;
+    if(batch_count < 0)
+    {
+        static const size_t        safe_size = 100; //  arbitrarily set to zero
+        device_vector<T>           dx(safe_size);
+        device_vector<rocblas_int> d_rocblas_result(std::abs(batch_count));
+        if(!dx || !d_rocblas_result)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+        EXPECT_ROCBLAS_STATUS((func(handle, N, dx, incx, stridex, batch_count, d_rocblas_result)),
+                              rocblas_status_invalid_size);
+        return;
+    }
+
+    // check to prevent undefined memory allocation error
+    if(N <= 0 || incx <= 0 || batch_count == 0)
+    {
+        static const size_t        safe_size = 100; //  arbitrarily set to zero
+        device_vector<T>           dx(safe_size);
+        device_vector<rocblas_int> d_rocblas_result(std::max(size_t(batch_count), safe_size));
+        if(!dx || !d_rocblas_result)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+        CHECK_ROCBLAS_ERROR((func(handle, N, dx, incx, stridex, batch_count, d_rocblas_result)));
+        return;
+    }
+
+    rocblas_int rocblas_result_1[batch_count], rocblas_result_2[batch_count],
+        cpu_result[batch_count];
+
+    size_t size_input = std::abs(stridex) * (batch_count - 1) + N * incx;
+
+    // allocate memory on device
+    device_vector<T>           dx(size_input);
+    device_vector<rocblas_int> d_rocblas_result_2(batch_count);
+    if(!dx || !d_rocblas_result_2)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Naming: dx is in GPU (device) memory. hx is in CPU (host) memory, plz follow this practice
+    host_vector<T> hx(size_input);
+
+    //
+    // Initialize data on CPU
+    //
+    rocblas_seedrand();
+    rocblas_init(hx, 1, N, incx, std::abs(stridex), batch_count);
+
+    // copy data from CPU to device, does not work for incx != 1
+    CHECK_HIP_ERROR(hipMemcpy(dx, hx, sizeof(T) * size_input, hipMemcpyHostToDevice));
+
+    T *ptrdx = (stridex > 0) ? ((T*)dx) : ((T*)dx) + (1 - batch_count) * stridex,
```

Comment:
I have chosen to be consistent with the convention ptrAk = ptrA0 + k * stride, coming from a discussion with Andrew.

Then, ptrdx and ptrhx are moved forward if stridex < 0.

Line 160, I'm able to move backward (consistent with the convention) because ptrhx has been moved forward before.

Reply:
In IAMIN/IAMAX, negative or zero increment means a zero-sized vector, and so it acts the same as `n<= 0`. When the dust settles, I'm going to make a sweep through BLAS1 and make sure it complies with expectations on `n` and increments, as well as early-exiting on success: If `n <= 0`, then `x != nullptr` does not need to be checked, and it can return `batch_count` copies of `0` (or `-1`) immediately with success, as long as `batch_count == 0 || (batch_count > 0 && result != nullptr)`

Reply:
If the BLAS function has a single vector, then a negative or zero vector increment is that same as negative or zero size, and the result is the mathematical result on a vector of size 0.

Negative increment conventions don't apply to strided batched strides, just problem increments.

Only when you have two vectors, or a vector and a matrix, does a negative vector increment go from the end.

If a vector is 0-sized, by its size being <= 0 or its single vector (no other vector or matrix) having an increment of <= 0, then the result is the mathematical result on a zero-sized vector, and the pointer to the vector can be `nullptr` without error. This is this "short-circuiting success" I'm talking about.